crates/oxc_linter/src/rules/no_bitwise.rs

```diff
@@ -0,0 +1,163 @@
+    int32_hint: bool,
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow bitwise operators
+    ///
+    /// ### Why is this bad?
+    ///
+    /// The use of bitwise operators in JavaScript is very rare and often `&` or `|` is simply a mistyped `&&` or `||`,
+    /// which will lead to unexpected behavior.
+    ///
+    /// ### Example
+    ///
+    /// ```javascript
+    /// var x = y | z;
+    /// ```
+    NoBitwise,
+    nursery
+);
+
+impl Rule for NoBitwise {
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let obj = value.get(0);
+
+        Self {
+            allow: obj
+                .and_then(|v| v.get("allow"))
+                .and_then(serde_json::Value::as_array)
+                .map(|v| {
+                    v.iter()
+                        .filter_map(serde_json::Value::as_str)
+                        .map(ToString::to_string)
+                        .collect()
+                })
+                .unwrap_or_default(),
+            int32_hint: obj
+                .and_then(|v| v.get("int32Hint"))
+                .and_then(serde_json::Value::as_bool)
+                .unwrap_or_default(),
+        }
+    }
+
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        match node.get().kind() {
+            AstKind::BinaryExpression(bin_expr) => {
+                let op = bin_expr.operator.as_str();
+
+                if bin_expr.operator.is_bitwise()
+                    && !allowed_operator(&self.allow, op)
+                    && !is_int32_hint(self.int32_hint, node)
+                {
+                    ctx.diagnostic(NoBitwiseDiagnostic(op, bin_expr.span));
+                }
+            }
+            AstKind::UnaryExpression(unary_expr) => {
+                let op = unary_expr.operator.as_str();
+
+                if unary_expr.operator.is_bitwise()
+                    && !allowed_operator(&self.allow, op)
+                    && !is_int32_hint(self.int32_hint, node)
+                {
+                    ctx.diagnostic(NoBitwiseDiagnostic(op, unary_expr.span));
+                }
+            }
+            AstKind::AssignmentExpression(assign_expr) => {
+                let op = assign_expr.operator.as_str();
+
+                if assign_expr.operator.is_bitwise()
+                    && !allowed_operator(&self.allow, op)
+                    && !is_int32_hint(self.int32_hint, node)
+                {
+                    ctx.diagnostic(NoBitwiseDiagnostic(op, assign_expr.span));
+                }
+            }
+            _ => {}
+        }
+    }
+}
+
+fn allowed_operator(allow: &[String], operator: &str) -> bool {
+    allow.iter().any(|s| s == operator)
+}
+
+fn is_int32_hint(int32_hint: bool, node: &AstNode) -> bool {
+    if !int32_hint {
+        return false;
+    }
+
+    match node.get().kind() {
+        AstKind::BinaryExpression(bin_expr) => {
+            bin_expr.operator.as_str() == "|" && bin_expr.right.is_number_0()
```

Comment:
You can match on the operator: `bin_expr.operator == BinaryOperator::BitwiseOR`.
We are not JavaScript ;-) we don't need to match on strings :-)

Reply:
done 😀